### PR TITLE
homework 2. Add simple rest service

### PR DESCRIPTION
**Pull request by captain2b**

### CODE REVIEW

File: hw-2/utils.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const getAutoSuggestUsers = (data = [], loginSubstring = '', limit = null) => {
```
Comment (farafonoff) [inline hw-2/utils.js]: As a homework, everything is OK, but in general, please avoid 'utils.js'. autoSuggestUsersService.js is much better name.

File: hw-2/utils.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  if (!loginSubstring && !limit) {
+    return data;
+  }
+  const filtered = data
+    // .sort()
```
Comment (farafonoff) [inline hw-2/utils.js]: sorting not implemented?

File: hw-2/server.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      cache.set(id, updated);
+      res.send(updated);
+    } else {
+      res.send(`User with id: ${id} not found`);
+    }
+  } catch (error) {
+    next(error);
+  }
+});
+router.post('/users/create', validator.body(createSchema), (req, res, next) => {
```
Comment (farafonoff) [inline hw-2/server.js]: doesn't follow REST best practices

File: hw-2/server.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    const { id } = req.params;
+    if (cache.has(id)) {
+      const updated = {
+        ...cache.get(id),
+        ...req.body,
+      };
+      cache.set(id, updated);
+      res.send(updated);
+    } else {
+      res.send(`User with id: ${id} not found`);
```
Comment (farafonoff) [inline hw-2/server.js]: doesn't follow REST best practices

Comment (captain2b) [reply hw-2/server.js]: changed to status 204 without a body. Is that ok?

Comment (farafonoff) [reply hw-2/server.js]: In general, put should do update or insert, but it depends on a project. On the other side, 204 is 'success' status, you shouldn't return it if request didn't work.

**Review by farafonoff — APPROVED** [review]

**Pull request merged**